src/main/java/uk/gov/register/core/Entry.java

```diff
@@ -80,13 +80,14 @@ public boolean equals(Object o) {
         Entry entry = (Entry) o;
 
         if (entryNumber != entry.entryNumber) return false;
-        return sha256hex == null ? entry.sha256hex == null : sha256hex.equals(entry.sha256hex);
+        return hashValue == null ? entry.hashValue == null : hashValue.getValue().equals(entry.hashValue.getValue());
     }
 
     @Override
     public int hashCode() {
-        int result = sha256hex != null ? sha256hex.hashCode() : 0;
+        String hash = hashValue.getValue();
```

Comment:
Similarly, you should not use `hashValue.getValue()` since you have implemented `HashValue.hashCode()`.